speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp

```diff
@@ -1,229 +1,487 @@
-#include <iostream>
 #include "SQLDatabaseStorage.h"
-#include "sqlite3.h"
 #include <string>
-#include <sstream>
+#include "mysqlx/xdevapi.h"
 #include "../rppUtils/RppUtils.hpp"
 
-/// @brief        Checks if the database threw an error
-/// @param  error The error that needs to be checked
-#define DATABASE_THROW(error) CheckForDatabaseError(rc, database, (std::string &) error, databasePath, zErrMsg, input);
+/// @brief reads input from input file, unless EOF has been reached
+#define READ_INPUT(p_string) \
+    if (m_inputFile.eof()) throw std::exception("Reached end of file prematurely");\
+    m_inputFile >> p_string;
+
+/// @brief executes sql statement with a safety net
+#define EXECUTE(p_sql) \
+    try {m_statement->execute(p_sql);} \
```

Comment:
execute returns a boolean, I do not know if this tells you whether it succeeded or not, but if this try catch can be prevented then please remove it

Reply:
It returns whether or not a result was returned by the sql query. Returns true for SELECT statements, and false for INSERT or CREATE